lib/rules/no-magic-numbers.js

```diff
@@ -6,6 +6,7 @@
 "use strict";
 
 const { isNumericLiteral } = require("./utils/ast-utils");
+const MAX_ARRAY_LENGTH = 2 ** 32 - 1;
```

Comment:
Thanks for naming this so that it's not a magic number, but I wonder if even more context would be helpful here? MDN has some [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length) we can use for this (and refer to in the docs)

Reply:
Completely agree :+1:  I'll update the docs.

Did you also mean to add more explanation in the code? There is already something in the description for `function isArrayIndex()`

Reply:
It's also interesting that the rule itself wouldn't allow this line in its own code.

This was proposed in #12535

Reply:
Updated the docs and comments in the code. I hope this specifies the option well and that it's explained in an understandable way, all suggestions are welcome!

Also, is using `<sup></sup>` okay? Couldn't find any examples in other documents.

Reply:
> Also, is using <sup></sup> okay?

According to the [Eleventy documentation](https://www.11ty.dev/docs/languages/markdown/#default-options), it looks like it should work! I think we'll just want to double check the site after the next release. :)

Reply:
`<sup></sup>` 2<sup>32</sup> looks good on site :tada: 

https://eslint.org/docs/7.0.0/rules/no-magic-numbers#ignorearrayindexes